File: common/src/main/java/com/pluxity/device/dto/DeviceCategoryResponseWithoutChildren.java
package com.pluxity.device.dto;

import com.pluxity.device.entity.DeviceCategory;
import com.pluxity.file.dto.FileResponse;
import io.swagger.v3.oas.annotations.media.Schema;

public record DeviceCategoryResponseWithoutChildren(
        Long id,
        String name,
        Long parentId,
        FileResponse thumbnailFile,
        @Schema(description = "depth", example = "1") int depth) {

    public static DeviceCategoryResponseWithoutChildren from(DeviceCategory deviceCategory) {
        return new DeviceCategoryResponseWithoutChildren(
                deviceCategory.getId(),
                deviceCategory.getName(),
                deviceCategory.getParent() != null ? deviceCategory.getParent().getId() : null,
                null,
                deviceCategory.getDepth());
    }

    public static DeviceCategoryResponseWithoutChildren from(
            DeviceCategory deviceCategory, FileResponse iconFile) {
        return new DeviceCategoryResponseWithoutChildren(
                deviceCategory.getId(),
                deviceCategory.getName(),
                deviceCategory.getParent() != null ? deviceCategory.getParent().getId() : null,
                iconFile != null ? iconFile : FileResponse.empty(),
                deviceCategory.getDepth());
    }
}

File: common/src/main/kotlin/com/pluxity/device/dto/DeviceCategoryResponseWithoutChildren.kt
package com.pluxity.device.dto

import com.pluxity.device.entity.DeviceCategory
import com.pluxity.file.dto.FileResponse
import io.swagger.v3.oas.annotations.media.Schema

data class DeviceCategoryResponseWithoutChildren(
    val id: Long?,
    val name: String,
    val parentId: Long?,
    val thumbnailFile: FileResponse,
    @field:Schema(
        description = "depth",
        example = "1",
    )
    val depth: Int,
)

fun DeviceCategory.toDeviceCategoryResponseWithoutChildren(iconFile: FileResponse) =
    DeviceCategoryResponseWithoutChildren(
        id = this.id,
        name = this.name,
        parentId = this.parent?.id,
        thumbnailFile = iconFile,
        depth = this.depth,
    )
